lib/src/main/scala/org/typelevel/scalacoptions/ScalaVersion.scala

```diff
@@ -44,6 +44,17 @@ object ScalaVersion {
   val V3_3_0   = ScalaVersion(3, 3, 0)
   val V3_3_1   = ScalaVersion(3, 3, 1)
 
+  private val versionRegex = raw"""(\d+)\.(\d+)\.(\d+)""".r
```

Comment:
[Yak Shaving Alert] Can I suggest something like this? I think we will do another PR before either Scala 4 will come out or any minor/hotfix of any existing scala version will pass number 99.
```suggestion
  private val versionRegex = raw"""(2|3)\.(\d\d?)\.(\d\d?)""".r
```